src/peft/tuners/xlora/classifier.py

```diff
@@ -0,0 +1,255 @@
+
+            if config.enable_relu_and_dropout:
+                layers.append(nn.ReLU())
+                layers.append(nn.Dropout(p=config.xlora_dropout_p))
+
+            if config.layerwise_scalings:
+                last = nn.Linear(config.xlora_size, n_classes * n_layers, bias=bias_flag).to(device).to(dtype)
+            else:
+                last = nn.Linear(config.xlora_size, n_classes, bias=bias_flag).to(device).to(dtype)
+        else:
+            if self.config.xlora_depth <= 0:
+                raise ValueError("X-LoRA depth must be strictly positive.")
+
+            layers.append(nn.Linear(config.hidden_size, config.xlora_size, bias=bias_flag).to(device).to(dtype))
+
+            if config.enable_relu_and_dropout:
+                layers.append(nn.ReLU())
+                layers.append(nn.Dropout(p=config.xlora_dropout_p))
+
+            for _ in range(config.xlora_depth - 2):
+                layers.append(nn.Linear(config.xlora_size, config.xlora_size, bias=bias_flag).to(device).to(dtype))
+
+                if config.enable_relu_and_dropout:
+                    layers.append(nn.ReLU())
+                    layers.append(nn.Dropout(p=config.xlora_dropout_p))
+
+            if config.layerwise_scalings:
+                last = nn.Linear(config.xlora_size, n_classes * n_layers, bias=bias_flag).to(device).to(dtype)
+            else:
+                last = nn.Linear(config.xlora_size, n_classes, bias=bias_flag).to(device).to(dtype)
+        self.layers = nn.Sequential(*layers, last)
+
+    def forward(
+        self,
+        input_ids: Optional[torch.LongTensor] = None,
+        inputs_embeds: Optional[torch.FloatTensor] = None,
+        *args,
+        **kwargs,
+    ) -> torch.Tensor:
+        """
+        Using the hidden states of the model, predict `n_classes` LoRA alpha values. Sets the scalings.
+        """
+        if input_ids is not None:
+            batch_size = input_ids.shape[0]
+        else:
+            batch_size = inputs_embeds.shape[0]
+
+        if input_ids is not None:
+            seq_len = input_ids.shape[1]
+        else:
+            seq_len = inputs_embeds.shape[1]
+
+        # For type checking
+        model = self.model
+        with torch.no_grad():
+            with model.disable_adapter():
+                kwargs["output_hidden_states"] = True
+                kwargs["return_dict"] = True
+
+                result: ModelOutput = model.forward(
+                    *args,
+                    input_ids=input_ids,
+                    inputs_embeds=inputs_embeds,
+                    _xlora_classifier_inhibitor_flag=InhibitorFlagPayload(
+                        batch_size=batch_size,
+                        seq_len=seq_len,
+                        override_scaling_pass_value=self.override_scaling_pass_value,
+                    ),
+                    **kwargs,
+                )
+
+        hidden_states = result.hidden_states  # type: ignore
+
+        assert hidden_states is not None
```

Comment:
Let's raise a proper error.

Reply:
Actually, I've never had that cause any trouble, and the lines below would catch any None forcefully. I have removed that check entirely.